src/nupic/algorithms/KNNClassifier.py

```diff
@@ -301,6 +310,11 @@ def removeCategory(self, categoryToRemove):
 
 
   def _removeRows(self, rowsToRemove):
+    """
+    A list of row indices to remove. There are two caveats. First, this is
+    a potentially slow operation. Second, pattern indices will shift if
```

Comment:
It seems like the previous version for clearing out the partition IDs would have been faster. If this is a method that needs to be fast then I might worry about the change.